cpp/include/cudf/utilities/traits.hpp

```diff
@@ -591,74 +591,63 @@ constexpr inline bool is_nested(data_type type)
   return cudf::type_dispatcher(type, is_nested_impl{});
 }
 
-template <typename FromType, typename ToType>
-struct is_logically_castable_impl : std::false_type {
-};
-
-// Allow cast to same type
-template <typename Type>
-struct is_logically_castable_impl<Type, Type> : std::true_type {
-};
-
-#ifndef MAP_CASTABLE_TYPES
-#define MAP_CASTABLE_TYPES(Type1, Type2)                             \
-  template <>                                                        \
-  struct is_logically_castable_impl<Type1, Type2> : std::true_type { \
-  };                                                                 \
-  template <>                                                        \
-  struct is_logically_castable_impl<Type2, Type1> : std::true_type { \
-  };
-#endif
-
-// Allow cast between timestamp and integer representation
-MAP_CASTABLE_TYPES(cudf::timestamp_D, cudf::timestamp_D::duration::rep);
-MAP_CASTABLE_TYPES(cudf::timestamp_s, cudf::timestamp_s::duration::rep);
-MAP_CASTABLE_TYPES(cudf::timestamp_ms, cudf::timestamp_ms::duration::rep);
-MAP_CASTABLE_TYPES(cudf::timestamp_us, cudf::timestamp_us::duration::rep);
-MAP_CASTABLE_TYPES(cudf::timestamp_ns, cudf::timestamp_ns::duration::rep);
-// Allow cast between durations and integer representation
-MAP_CASTABLE_TYPES(cudf::duration_D, cudf::duration_D::rep);
-MAP_CASTABLE_TYPES(cudf::duration_s, cudf::duration_s::rep);
-MAP_CASTABLE_TYPES(cudf::duration_ms, cudf::duration_ms::rep);
-MAP_CASTABLE_TYPES(cudf::duration_us, cudf::duration_us::rep);
-MAP_CASTABLE_TYPES(cudf::duration_ns, cudf::duration_ns::rep);
-// Allow cast between decimals and integer representation
-MAP_CASTABLE_TYPES(numeric::decimal32, numeric::decimal32::rep);
-MAP_CASTABLE_TYPES(numeric::decimal64, numeric::decimal64::rep);
-
 template <typename FromType>
-struct is_logically_castable_to_impl {
-  template <typename ToType>
+struct is_bit_castable_to_impl {
+  template <typename ToType,
+            typename std::enable_if_t<cuda::std::is_same_v<ToType, cudf::string_view>>* = nullptr>
+  constexpr bool operator()()
+  {
+    return false;
+  }
+
+  template <
+    typename ToType,
+    typename std::enable_if_t<not cuda::std::is_same_v<ToType, cudf::string_view>>* = nullptr>
   constexpr bool operator()()
   {
-    return is_logically_castable_impl<FromType, ToType>::value;
+    if (not cuda::std::is_trivially_copyable_v<FromType> ||
+        not cuda::std::is_trivially_copyable_v<ToType>) {
+      return false;
+    }
+    constexpr auto from_size = sizeof(cudf::device_storage_type_t<FromType>);
+    constexpr auto to_size   = sizeof(cudf::device_storage_type_t<ToType>);
+    return from_size == to_size;
   }
 };
 
-struct is_logically_castable_from_impl {
-  template <typename FromType>
+struct is_bit_castable_from_impl {
+  template <typename FromType,
+            typename std::enable_if_t<cuda::std::is_same_v<FromType, cudf::string_view>>* = nullptr>
+  constexpr bool operator()(data_type)
+  {
+    return false;
+  }
```

Comment:
When you're only filtering for a single type, an explicit specialization is much cleaner than an `enable_if`.

Reply:
I propose to change this to `cudf::is_compound<FromType>` actually.

Reply:
` is_compound` check sound suitable for this scenario.